src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Utils/SPDXToSbomFormatConverterExtensions.cs

```diff
@@ -15,36 +18,186 @@ namespace Microsoft.Sbom.Utils;
         return new SbomFile
         {
-            Checksum = checksums,
-            FileCopyrightText = spdxFile.CopyrightText,
+            Checksum = spdxFile.VerifiedUsing.ToSbomChecksum(),
+            FileCopyrightText = spdxFile.CopyrightText == "NOASSERTION" ? null : spdxFile.CopyrightText,
+            Path = spdxFile.Name,
             Id = spdxFile.SpdxId,
-            Path = spdxFile.Name
+            LicenseConcluded = spdx30Elements != null && relationships != null
+            ? spdxFile.GetSingleLicense(RelationshipType.HAS_CONCLUDED_LICENSE, spdx30Elements, relationships)
+            : null,
+            LicenseInfoInFiles = spdx30Elements != null && relationships != null
+            ? spdxFile.GetMultipleLicenses(RelationshipType.HAS_DECLARED_LICENSE, spdx30Elements, relationships)
+            : null
+        };
+    }
+
+    public static SbomPackage ToSbomPackage(this Package spdxPackage, List<Element> spdx30Elements, List<Relationship> relationships)
+    {
+        var sbomPackage = new SbomPackage
+        {
+            PackageName = spdxPackage.Name,
+            PackageVersion = spdxPackage.PackageVersion,
+            PackageSource = spdxPackage.DownloadLocation == "NOASSERTION" ? null : spdxPackage.DownloadLocation,
+            CopyrightText = spdxPackage.CopyrightText == "NOASSERTION" ? null : spdxPackage.CopyrightText,
+            Checksum = spdxPackage.VerifiedUsing.ToSbomChecksum(),
+            LicenseInfo = new LicenseInfo
+            {
+                Concluded = spdxPackage.GetSingleLicense(RelationshipType.HAS_CONCLUDED_LICENSE, spdx30Elements, relationships),
+                Declared = spdxPackage.GetSingleLicense(RelationshipType.HAS_DECLARED_LICENSE, spdx30Elements, relationships),
+            },
+            Supplier = spdxPackage.GetSupplier(spdx30Elements),
+            PackageUrl = spdxPackage.GetPackageUrl(spdx30Elements),
+            FilesAnalyzed = true,
+            Id = spdxPackage.SpdxId,
+        };
+
+        return sbomPackage;
+    }
+
+    public static List<SbomRelationship> ToSbomRelationship(this Relationship spdxRelationship)
+    {
+        var sbomRelationships = new List<SbomRelationship>();
+        foreach (var toElement in spdxRelationship.To)
+        {
+            sbomRelationships.Add(
+            new SbomRelationship
+            {
+                SourceElementId = spdxRelationship.From,
+                RelationshipType = spdxRelationship.RelationshipType.ToString(),
+                TargetElementId = toElement,
+            });
+        }
+
+        return sbomRelationships;
+    }
+
+    public static SbomReference ToSbomReference(this ExternalMap externalDocumentReference)
+    {
+        if (externalDocumentReference is null)
+        {
+            return null;
+        }
+
+        return new SbomReference
+        {
+            Checksum = externalDocumentReference.VerifiedUsing?.ToSbomChecksum().FirstOrDefault(),
```

Comment:
Do we ever expect there to be more than one here? Why FirstOrDefault?

Reply:
There will never be more than one. In the public facing SPDX documentation, VerifiedUsing is always represented as a list. However, internally for SbomReference's there's only one allowed checksum value.